crates/sui-json-rpc/src/unit_tests/rpc_server_tests.rs

```diff
@@ -510,6 +512,64 @@ async fn test_get_transaction() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[sim_test]
+async fn test_get_checkpoints() -> Result<(), anyhow::Error> {
+    let cluster = TestClusterBuilder::new().build().await?;
+    let http_client = cluster.rpc_client();
+    let address = cluster.accounts.first().unwrap();
+
+    let objects = http_client.get_objects_owned_by_address(*address).await?;
+    let gas_id = objects.last().unwrap().object_id;
+
+    // Make some transactions
+    let mut tx_responses: Vec<SuiTransactionResponse> = Vec::new();
+    for oref in &objects[..objects.len() - 1] {
+        let transaction_bytes: TransactionBytes = http_client
+            .transfer_object(*address, oref.object_id, Some(gas_id), 1000, *address)
+            .await?;
+        let keystore_path = cluster.swarm.dir().join(SUI_KEYSTORE_FILENAME);
+        let keystore = Keystore::from(FileBasedKeystore::new(&keystore_path)?);
+        let tx =
+            to_sender_signed_transaction(transaction_bytes.to_data()?, keystore.get_key(address)?);
+
+        let (tx_bytes, signatures) = tx.to_tx_bytes_and_signatures();
+
+        let response = http_client
+            .execute_transaction(
+                tx_bytes,
+                signatures,
+                Some(SuiTransactionResponseOptions::new()),
+                Some(ExecuteTransactionRequestType::WaitForLocalExecution),
+            )
+            .await?;
+
+        tx_responses.push(response);
+    }
+
+    sleep(Duration::from_millis(6000)).await;
```

Comment:
@lxfind @patrickkuo do you know how can we make this number smaller(e.g., to get a checkpoint sooner)?

Alternatively @siomari , I think it would be easier to add this test to `checkpoint.test.ts`